scene/gui/popup.cpp

```diff
@@ -75,16 +75,17 @@ void Popup::_fix_size() {
 	Point2 pos = get_global_position();
 	Size2 size = get_size() * get_scale();
 	Point2 window_size = get_viewport_rect().size;
+	Point2 vpos = get_viewport_transform().get_origin();
 
-	if (pos.x + size.width > window_size.width)
-		pos.x = window_size.width - size.width;
-	if (pos.x < 0)
-		pos.x = 0;
+	if (pos.x + size.width > window_size.width - vpos.x)
```

Comment:
You should get the same by simply subtracting `vpos` to `pos` at the end of the calculations instead of doing it separately in all checks and coordinate assignments, no?

Reply:
Yes, you're right... however by window_size, not pos.